src/controller/CHIPDeviceController.cpp

```diff
@@ -1889,12 +1891,23 @@ void DeviceCommissioner::OnScanNetworksResponse(void * context,
                                                : "none provided"));
     DeviceCommissioner * commissioner = static_cast<DeviceCommissioner *>(context);
 
+    // advance to the kNeedsNetworkCreds waiting step
+    commissioner->CommissioningStageComplete(CHIP_NO_ERROR);
+
     if (commissioner->GetPairingDelegate() != nullptr)
```

Comment:
Now that we are going to kNeedsNetworkCreds, we have no choice but to have a pairing delegate. This is a change of public interface contract --> it was not necessary before to have the pairing delegate always notify of credentials success. 

This may break external users of the SDK who did not have these methods do anything to advance the state machine.